### DISCUSSION TITLE
Provide a duration object in addition to ms

### DISCUSSION THREAD

**Discussion by Liam-Tait** [TanStack/query#2327] [Ideas] [answered]
In the examples it is common to annotate the intended duration with a comment next to some multiplication for determining the duration

```ts
import { useQuery } from "react-query";
const useTodos = () =>
  useQuery("todos", () => fetch("/todos"), {
    staleTime: 1000 * 60 * 5, // 5 minutes
  });
```

I would like to suggest adding a Duration object as an option

```ts
const useTodos = () =>
  useQuery("todos", () => fetch("/todos"), {
    staleTime: { minutes: 5 },
  });
```

This Duration format is inspired by [date-fns Duration](https://date-fns.org/v2.22.1/docs/Duration)

The typing could be the following

```ts
type Duration = {
  years?: number;
  months?: number;
  weeks?: number;
  days?: number;
  hours?: number;
  minutes?: number;
  seconds?: number;
};
```

I believe this would be in scope of `react-query` as it prevents errors from happening in a common usecase and self documents intention.

For my use case it is common to set `staleTime` or `refetchInterval` to X minutes. I am interested to know if this would be useful to people/

This would be a non-breaking change as Duration can continue to treated as `ms` when a number is provided

**Comment by TkDodo** [comment, marked as answer]
I'd rather leave that to user-land, as there is no standardized way of doing it. If you are using [dayjs](https://day.js.org/) for example, you can already do it via:

```
staleTime: dayjs.duration({ minutes: 5 }).asMilliseconds()
```
